inspirehep/modules/workflows/workflows/edit_article.py

```diff
@@ -41,6 +47,19 @@ def update_record(obj, eng):
     record.update(obj.data)
     record.commit()
 
+    transaction_id = None
+    record = get_db_record('lit', control_number)
+    for revision in record.revisions:
+        if revision.revision_id == record.revision_id:
+            transaction_id = revision.model.transaction_id
+            break
```

Comment:
maybe something like:

```python
try:
  revision = record.revision[record.revision_id]
  transaction_id = revision.model.transaction_id
expect IndexError:
  pass # do something
```

Reply:
I'll need to verify, that thing is not an array index or not.

Reply:
you should never index into `record.revisions`. It's completely broken, as `record == record.revisions[record.revision_id]` does not hold (because a new `revision_id` might be generated but not committed, so it can jump).